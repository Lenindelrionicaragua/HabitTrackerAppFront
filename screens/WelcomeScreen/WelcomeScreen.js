import React from "react";
import { StatusBar } from "react-native";

import {
  StyledContainer,
  InnerContainer,
  PageTitle,
  SubTitle,
  StyledFormArea,
  StyledButton,
  ButtonText,
  Line,
  WelcomeContainer,
  WelcomeImage,
  Avatar
} from "./WelcomeScreenStyles";

const WelcomeScreen = ({ navigation, route }) => {
  const { name, email } = route.params;

  return (
    <StyledContainer testID="styled-container">
      <StatusBar style="light" />
      <InnerContainer testID="inner-container">
        <WelcomeImage
          resizeMode="cover"
          source={require("./../../assets/ZenTimer6.png")}
          testID="welcome-image"
        />

        <WelcomeContainer testID="welcome-container">
          <PageTitle welcome={true} testID="welcome-title">
            Welcome!
          </PageTitle>
          <SubTitle welcome={true} testID="user-name">
            {name || "Zen User"}
          </SubTitle>
          <SubTitle welcome={true} testID="user-email">
            {email || "serenity@gmail.com"}
          </SubTitle>
          <StyledFormArea>
            <Avatar
              resizeMode="cover"
              source={require("./../../assets/logoZenTimer.png")}
              testID="avatar-image"
            />
            <Line testID="line" />
            <StyledButton
              onPress={() => {
                navigation.navigate("LoginScreen");
              }}
              testID="logout-styled-button"
            >
              <ButtonText testID="logout-button-text">Logout</ButtonText>
            </StyledButton>
          </StyledFormArea>
        </WelcomeContainer>
      </InnerContainer>
    </StyledContainer>
  );
};

export default WelcomeScreen;
