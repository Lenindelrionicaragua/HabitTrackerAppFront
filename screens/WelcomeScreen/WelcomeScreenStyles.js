import styled from "styled-components/native";
import { View, Image, Text, Pressable } from "react-native";
import { SafeAreaView } from "react-native";
import { Colors } from "../../styles/AppStyles";

const { white, orange, grey, yellow, lightGrey, black } = Colors;

export const StyledContainer = styled(SafeAreaView)`
  flex: 1;
  background-color: ${lightGrey};
  width: 100%;
`;

export const InnerContainer = styled(View)`
  flex: 1;
  width: 100%;
  align-items: center;
  justify-content: center;
`;

export const WelcomeContainer = styled(InnerContainer)`
  padding: 25px;
  padding-top: 10px;
  justify-content: center;
`;

export const Avatar = styled(Image)`
  width: 100px;
  height: 100px;
  margin: auto;
  border-radius: 50px;
  border-width: 2px;
  border-color: ${yellow};
  margin-bottom: 10px;
  margin-top: 10px;
`;

export const WelcomeImage = styled(Image)`
  height: 50%;
  width: 100%;
`;

export const PageTitle = styled(Text)`
  font-size: 30px;
  text-align: center;
  font-weight: bold;
  color: ${black};
  padding: 10px;

  ${props =>
    props.welcome &&
    `
    font-size: 35px;
  `}
`;

export const SubTitle = styled(Text)`
  font-size: 15px;
  margin-bottom: 20px;
  letter-spacing: 1px;
  font-weight: bold;
  color: ${grey};

  ${props =>
    props.welcome &&
    `
    margin-bottom: 5px;
    font-weight: normal;
  `}
`;

export const StyledFormArea = styled(View)`
  width: 90%;
`;

export const StyledButton = styled(Pressable)`
  padding: 15px;
  background-color: ${black};
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  margin-vertical: 5px;
  height: 60px;

  ${props =>
    props.google == true &&
    `
    background-color: ${orange};
    flex-direction: row;
    justify-content: center;
  `}
`;

export const ButtonText = styled(Text)`
  color: ${white};
  font-size: 16px;

  ${props =>
    props.google == true &&
    `
    padding: 25px;
    color: ${black}
  `}
`;

export const Line = styled(View)`
  height: 1px;
  width: 100%;
  background-color: ${orange};
  margin-vertical: 10px;
`;
