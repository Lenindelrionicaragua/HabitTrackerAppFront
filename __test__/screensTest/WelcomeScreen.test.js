import React from "react";
import renderer from "react-test-renderer";
import { render, cleanup, fireEvent, act } from "@testing-library/react-native";
import MockAdapter from "axios-mock-adapter";
import axios from "axios";

import WelcomeScreen from "../../screens/WelcomeScreen/WelcomeScreen";

import { StatusBar } from "react-native";

// Rendering Functions
const renderWelcomeScreen = routeParams =>
  render(<WelcomeScreen route={{ params: routeParams }} />);
const renderWelcomeScreenWithRenderer = routeParams =>
  renderer.create(<WelcomeScreen route={{ params: routeParams }} />);

let welcomeScreenRender;
let welcomeScreenRenderWithRenderer;

beforeEach(() => {
  welcomeScreenRender = renderWelcomeScreen({
    name: "Zen User",
    email: "serenity@gmail.com"
  });
  welcomeScreenRenderWithRenderer = renderWelcomeScreenWithRenderer({
    name: "Zen User",
    email: "serenity@gmail.com"
  });
});

//SignupScreen
describe("WelcomeScreen", () => {
  afterEach(() => {
    cleanup();
  });

  test("Renders correctly the WelcomeScreen Component", () => {
    const WelcomeScreenSnapshot = welcomeScreenRenderWithRenderer.toJSON();
    expect(WelcomeScreenSnapshot).toMatchSnapshot();
  });

  test("Render StatusBar correctly", () => {
    const WelcomeScreenInstance = welcomeScreenRenderWithRenderer.root;
    const statusBar = WelcomeScreenInstance.findByType(StatusBar);
    expect(statusBar).toBeTruthy();
  });

  test("Render UI components correctly", () => {
    const { getByTestId } = welcomeScreenRender;
    expect(getByTestId("inner-container")).toBeTruthy();
    expect(getByTestId("welcome-container")).toBeTruthy();
    expect(getByTestId("welcome-title")).toBeTruthy();
    expect(getByTestId("user-name")).toBeTruthy();
    expect(getByTestId("user-email")).toBeTruthy();
  });

  test("PageTitle should render a string of letters, numbers, or spaces", () => {
    const { getByTestId } = welcomeScreenRender;
    const pageTitleComponent = getByTestId("welcome-title");
    const textContent = pageTitleComponent.props.children.toString();
    expect(textContent).toMatch("Welcome!");
  });

  test("SubTitle should render a string of letters, numbers or spaces", () => {
    const { getByTestId } = welcomeScreenRender;
    const subTitleComponent = getByTestId("user-name");
    const textContent = subTitleComponent.props.children.toString();
    expect(textContent).toMatch("Zen User");
  });

  test("SubTitle should render a string of letters, numbers or spaces", () => {
    const { getByTestId } = welcomeScreenRender;
    const subTitleComponent = getByTestId("user-email");
    const textContent = subTitleComponent.props.children.toString();
    expect(textContent).toMatch("serenity@gmail.com");
  });
});

// WelcomeImage
describe("WelcomeImage", () => {
  afterEach(() => {
    cleanup();
  });

  test("Render the WelcomePage component correctly and has a valid image source", () => {
    const { getByTestId } = welcomeScreenRender;
    const welcomeImageComponent = getByTestId("welcome-image");
    expect(welcomeImageComponent).toBeTruthy();
  });
});

// Avatar
describe("AvatarImage", () => {
  afterEach(() => {
    cleanup();
  });

  test("Render the AvatarImage component correctly and has a valid image source", () => {
    const { getByTestId } = welcomeScreenRender;
    const avatarImageComponent = getByTestId("avatar-image");
    expect(avatarImageComponent).toBeTruthy();
  });
});

// Logout ButtonText
describe("Logout ButtonText", () => {
  afterEach(() => {
    cleanup();
  });

  test("Renders ButtonText with the correct text", () => {
    const { getByTestId } = welcomeScreenRender;
    const styledButtonElement = getByTestId("logout-button-text");
    const textContent = styledButtonElement.toString();
    expect(styledButtonElement).toBeTruthy();
    expect(textContent).toMatchSnapshot("Logout");
  });
});

// Navigation Test

describe("WelcomeScreen navigation", () => {
  let navigation;

  beforeEach(() => {
    navigation = { navigate: jest.fn() };
    welcomeScreenRender = render(
      <WelcomeScreen navigation={navigation} route={{ params: {} }} />
    );
  });

  afterEach(() => {
    cleanup();
  });

  test("navigate to LoginScreen when Logout button is clicked", () => {
    const { getByTestId } = welcomeScreenRender;
    const styledButtonElement = getByTestId("logout-styled-button");
    act(() => {
      fireEvent.press(styledButtonElement);
    });
    expect(navigation.navigate).toHaveBeenCalledWith("LoginScreen");
  });
});
