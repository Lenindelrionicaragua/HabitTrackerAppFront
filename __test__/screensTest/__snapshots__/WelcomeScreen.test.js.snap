// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout ButtonText Renders ButtonText with the correct text: Logout 1`] = `"[object Object]"`;

exports[`WelcomeScreen Renders correctly the WelcomeScreen Component 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#f9f7f3",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "width": "100%",
    }
  }
  testID="styled-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="inner-container"
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        {
          "height": "50%",
          "width": "100%",
        }
      }
      testID="welcome-image"
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 25,
          "paddingLeft": 25,
          "paddingRight": 25,
          "paddingTop": 10,
          "width": "100%",
        }
      }
      testID="welcome-container"
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 35,
            "fontWeight": "bold",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "textAlign": "center",
          }
        }
        testID="welcome-title"
        welcome={true}
      >
        Welcome!
      </Text>
      <Text
        style={
          {
            "color": "#1a1204",
            "fontSize": 15,
            "fontWeight": "normal",
            "letterSpacing": 1,
            "marginBottom": 5,
          }
        }
        testID="user-name"
        welcome={true}
      >
        Zen User
      </Text>
      <Text
        style={
          {
            "color": "#1a1204",
            "fontSize": 15,
            "fontWeight": "normal",
            "letterSpacing": 1,
            "marginBottom": 5,
          }
        }
        testID="user-email"
        welcome={true}
      >
        serenity@gmail.com
      </Text>
      <View
        style={
          {
            "width": "90%",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={1}
          style={
            {
              "borderBottomLeftRadius": 50,
              "borderBottomRightRadius": 50,
              "borderColor": "#ffc554",
              "borderTopLeftRadius": 50,
              "borderTopRightRadius": 50,
              "borderWidth": 2,
              "height": 100,
              "marginBottom": 10,
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 10,
              "width": 100,
            }
          }
          testID="avatar-image"
        />
        <View
          style={
            {
              "backgroundColor": "#ffc500",
              "height": 1,
              "marginVertical": 10,
              "width": "100%",
            }
          }
          testID="line"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "height": 60,
              "justifyContent": "center",
              "marginVertical": 5,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
          testID="logout-styled-button"
        >
          <Text
            style={
              {
                "color": "#ffe1a8",
                "fontSize": 16,
              }
            }
            testID="logout-button-text"
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
